contrib/externs/chrome.js

```diff
@@ -76,29 +76,170 @@ function ChromeEvent() {}
 /**
  * @param {!Function} callback
  * @return {undefined}
+ * @see https://developer.chrome.com/extensions/events#method-Event-addListener
  */
 ChromeEvent.prototype.addListener = function(callback) {};
 
 
 /**
  * @param {!Function} callback
  * @return {undefined}
+ * @see https://developer.chrome.com/extensions/events#method-Event-removeListener
  */
 ChromeEvent.prototype.removeListener = function(callback) {};
 
 
 /**
  * @param {!Function} callback
  * @return {boolean}
+ * @see https://developer.chrome.com/extensions/events#method-Event-hasListener
  */
 ChromeEvent.prototype.hasListener = function(callback) {};
 
 
-/** @return {boolean} */
+/**
+ * @return {boolean}
+ * @see https://developer.chrome.com/extensions/events#method-Event-hasListeners
+ */
 ChromeEvent.prototype.hasListeners = function() {};
 
-// TODO(tbreisacher): Add the addRules, getRules, and removeRules methods?
 
+/**
+ * @param {!Array<!Rule>} rules
+ * @param {function(!Array<!Rule>): void=} callback
+ * @see https://developer.chrome.com/extensions/events#method-Event-addRules
+ */
+ChromeEvent.prototype.addRules = function(rules, callback) {};
+
+
+/**
+ * @param {!Array<string>|undefined} ruleIdentifiers
+ * @param {!function(!Array<!Rule>): void} callback
+ * @see https://developer.chrome.com/extensions/events#method-Event-getRules
+ */
+ChromeEvent.prototype.getRules = function(ruleIdentifiers, callback) {};
+
+
+/**
+ * @param {Array<string>=} ruleIdentifiers
+ * @param {function(): void=} callback
+ * @see https://developer.chrome.com/extensions/events#method-Event-removeRules
+ */
+ChromeEvent.prototype.removeRules = function(ruleIdentifiers, callback) {};
+
+
+/**
+ * @see https://developer.chrome.com/extensions/events#type-Rule
+ * @constructor
```

Comment:
To get the behavior you want, this needs to be `@record`

I'm looking at the example here:
https://developer.chrome.com/extensions/declarativeContent#type-Rule

It creates rules by just defining objects with the appropriately named properties.
That's what `@record` gets you. The compiler will consider any object with the right properties to have the type.

With `@constructor` the compiler expects to see `new Rule()` or a call to some function that returns a `Rule` in order to get one.

More info here
https://github.com/google/closure-compiler/wiki/Structural-Interfaces-in-Closure-Compiler
and here
https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#interface-record

Reply:
@brad4d oh, same goes to `UrlFilter` then, I will change that.

Reply:
@brad4d done 👍